### PR TITLE
Add Overlay Mouse, Lightgun, and Pointer

### LINKED ISSUE #15452: ds bug
when holding a button on a gamepad overlay, you cant use the ds touchscreen while holding a button. touches do nothing. using melonDS core on iphone 12 pro. latest retroarch nightly build.

### PR DESCRIPTION
**Pull request by neil4**
# Add Overlay Mouse, Lightgun, and Pointer

This creates mouse, lightgun, and pointer input as the overlay is polled. When enabled, touch inputs not in a hitbox are assumed to be pointing device input for the core.

The main benefit is that overlay buttons can be used simultaneously, so things like mouse-controlled 1st-person shooters and touchscreen emulation are more natural.

## Mouse behavior
1/2/3-finger taps are LMB/RMB/MMB clicks. A 0.2s long press begins holding a button, and you'll feel haptic feedback when it starts. A double-tap hold method is also available but adds latency to clicks. Dragging 2/3 fingers immediately holds the RMB/MMB.

There are settings for mouse speed, long-press and double-tap time thresholds, and a "swipe threshold" to distinguish swipes from taps and long presses.

## Lightgun behavior
Lightgun x, y, and trigger are normally sent together to every lightgun port. There are settings to choose the port, disable the trigger, delay the trigger, and clamp off-screen aim to the in-bounds edge. The trigger delay is needed for content that doesn't instantly move the gun cursor to a screen tap.

More lightgun buttons can be assigned to 2/3/4-finger inputs. The trigger delay is used to wait for the correct multi-touch count. A 1-frame delay is usually enough to distinguish these inputs.

### Notes
 - A few cores have a "touchscreen" option for lightgun input. That needs to be set to "lightgun" for this. I expect some confusion over that.
 - This does not implement the deprecated relative x/y IDs, and I haven't seen any problems from it.

## Related Pull Requests
The lite/SNES and lite/SNES_2 overlays in libretro/common-overlays#111 each include a lightgun overlay with buttons on the perimeter.

Worth a note: The existing lite/DOS and lite/DOS_2 overlays are already set up for 1st-person shooters, since each can hide the ABXY buttons.

## Related Issues
Fixes #15452 #15749 #15973

### CODE REVIEW

**Comment by hizzlekizzle** [comment]
This sounds great. We should probably copy this commit description to the docs site, as it will get lost in the shuffle here, and there's a lot of obscure knowledge (like the multi-finger tap commands) that users will never figure out on their own.

**Comment by neil4** [comment]
Sure. I don't think it's so obscure, but then I wrote it. I'll find a place to add this in the docs.

File: input/input_driver.c [hunk 1]
```diff
@@ -2716,6 +2893,285 @@ void input_overlay_auto_rotate_(
...
+ * @settings: pointer to settings
+ * @ol : overlay handle
+ * @old_ptr_count : previous poll's non-hitbox pointer count
+ *
+ * Updates multi-touch button state of the overlay lightgun.
+ */
+static void input_overlay_poll_lightgun(settings_t *settings,
+      input_overlay_t *ol, const int old_ptr_count)
+{
+   input_overlay_state_t *ol_state       = &ol->overlay_state;
```
Comment (warmenhoven) [inline input/input_driver.c:2907]: This variable is unused

**Comment by warmenhoven** [comment]
The combined overlay/touchscreen input is working on iOS as I would have expected. Very nice, thanks!

**Comment by neil4** [comment]
@warmenhoven Oh thanks, iOS is always the platform I wish I could test with this kind of thing

**Comment by neil4** [comment]
It looks like I should explain the mouse buttons in a sublabel, and remove an unused variable.

In the docs I only saw the touchscreen mouse explained for the Switch, and it's very different from Android's which this PR roughly follows.

**Pull request merged**